core/services/job_subscriber.go

```diff
@@ -24,12 +24,14 @@ type jobSubscriber struct {
 	store            *store.Store
 	jobSubscriptions map[string]JobSubscription
 	jobsMutex        *sync.RWMutex
+	runManager       RunManager
 }
 
 // NewJobSubscriber returns a new job subscriber.
-func NewJobSubscriber(store *store.Store) JobSubscriber {
+func NewJobSubscriber(store *store.Store, runManager RunManager) JobSubscriber {
 	return &jobSubscriber{
 		store:            store,
+		runManager:       runManager,
```

Comment:
Should the runManager be a pointer for the JobSubscriber? I'm not sure we want to make copies of it. Wondering the same thing about the RunQueue that the manager holds, but don't really know how copying the `map` in there behaves.

Reply:
Interfaces are implicitly pointers.